api/v1beta1/disruption_schedule_types.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func init() {
+	SchemeBuilder.Register(&DisruptionSchedule{}, &DisruptionScheduleList{})
+}
+
+//+kubebuilder:object:root=true
+
+// DisruptionSchedule is the Schema for the disruptions API
+// +kubebuilder:resource:shortName=disch
+// +kubebuilder:subresource:status
+type DisruptionSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              DisruptionScheduleSpec   `json:"spec,omitempty"`
+	Status            DisruptionScheduleStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// DisruptionScheduleList contains a list of DisruptionSchedule
+type DisruptionScheduleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DisruptionSchedule `json:"items"`
+}
+
+type DisruptionScheduleSpec struct {
+	//+kubebuilder:validation:MinLength=0
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule string `json:"schedule"`
+
+	// TargetResource specifies the resource to run disruptions against.
+	// It can only be a Deployment or StatefulSet.
+	TargetResource TargetResource `json:"targetResource"`
+
+	// Specifies the Disruption that will be created when executing a DisruptionShedule.
+	DisruptionTemplate Disruption `json:"disruptionTemplate"`
+}
+
+// TargetResource specifies the long-lived resource to be targeted for disruptions.
+// Disruptions are intended to exist semi-permanently, and thus appropriate targets can only be other long-lived resources,
+// such as StatefulSets or Deployments.
+type TargetResource struct {
+	// Kind specifies the type of the long-lived resource. Allowed values: "Deployment", "StatefulSet".
+	// +kubebuilder:validation:Enum=Deployment;StatefulSet
+	Kind string `json:"kind"`
+
+	// Name specifies the name of the specific instance of the long-lived resource to be targeted.
+	Name string `json:"name"`
+}
+
+type DisruptionScheduleStatus struct {
+	// The last time when the disruption was last successfully scheduled.
+	// +optional
+	LastScheduleTime *metav1.Time `json:"lastScheduleTime,omitempty"`
+
+	// Time when the target resource was previously missing.
```

Comment:
Why not choose a more generic field here, such as last failed schedule, given the other field is last successful schedule. This doesn't leave us space in the status to show failed schedules for reasons other than missing targets

Reply:
You may even want a field like "lastFailedSchedule" that includes a time _and_ a `reason` string field

Reply:
Okay, I now see why we want this specific PreviouslyMissing field, to handle cleanup of an orphaned disruption. I take back my suggestion